File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int Sum = 0;
        int Start = 0; 
        double average = 0;   
        for(int i = 0; i < k; i++){
            Sum += nums[i];
        }

         average = (double) Sum/k; 

        for(int i = k; i < nums.size(); i++){
            Sum += nums[i] - nums[i - k];  
            double average2 = (double)Sum /k; 

            if(average2 > average){
                average = average2; 
            }
            

            
        }

        return average; 
        
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        left = 0
        Sum = sum(nums[left:k])
        max_avg = Sum
        for r in range(k, len(nums)):
            Sum -= nums[left]
            Sum += nums[r]
            
            max_avg = max(max_avg, Sum)
            left += 1

        return max_avg/k